### PR TITLE
Candles Cache Improvements

### PR DESCRIPTION
**Pull request by adamfraser**
Additions: 
- An orderbook mid price cache that holds a series of prices, any price older that 30 seconds is removed from the cache
- Functionality to retrieve the median price from the series stored in the new orderbook mid prices cache
- A roundtable job that fetches the current orderbook mid price for all markets and adds them to the new cache, this runs every 10 seconds
- An in memory cache in ender that fetches the latest median price every 10 seconds 
-  Candles orderbook mid price open / close values are set using the median prices stored in ender cache

## Summary by CodeRabbit

- **New Features**
	- Introduced caching for order book mid prices, enhancing data retrieval efficiency.
	- Added configuration options to control caching behavior and update intervals.
	- New Lua scripts for managing and retrieving mid prices in Redis.

- **Bug Fixes**
	- Improved error handling for undefined prices in caching tasks.

- **Tests**
	- Expanded test coverage for order book mid price caching and candle generation processes.

- **Documentation**
	- Updated comments and documentation to reflect new functionalities and configurations.

### CODE REVIEW

File: .github/workflows/indexer-build-and-push-dev-staging.yml [hunk 1]
```diff
@@ -6,53 +6,14 @@ on: # yamllint disable-line rule:truthy
       - main
       - 'release/indexer/v[0-9]+.[0-9]+.x'  # e.g. release/indexer/v0.1.x
       - 'release/indexer/v[0-9]+.x'  # e.g. release/indexer/v1.x
+      - 'adam/candles-cache-main'
```
Comment (vincentwschau) [inline .github/workflows/indexer-build-and-push-dev-staging.yml:9]: nit: Should remove before merging.

File: .github/workflows/indexer-build-and-push-mainnet.yml [hunk 2]
```diff
@@ -6,6 +6,7 @@ on: # yamllint disable-line rule:truthy
       - main
       - 'release/indexer/v[0-9]+.[0-9]+.x'  # e.g. release/indexer/v0.1.x
       - 'release/indexer/v[0-9]+.x'  # e.g. release/indexer/v1.x
+      - 'adam/candles-cache-main'
```
Comment (vincentwschau) [inline .github/workflows/indexer-build-and-push-mainnet.yml:9]: nit: Remove before merging.

File: indexer/services/roundtable/src/config.ts [hunk 3]
```diff
@@ -138,7 +139,7 @@ export const configSchema = {
     default: 30 * ONE_DAY_IN_MILLISECONDS,
   }),
   LOOPS_INTERVAL_MS_CACHE_ORDERBOOK_MID_PRICES: parseInteger({
-    default: ONE_SECOND_IN_MILLISECONDS,
+    default: ONE_SECOND_IN_MILLISECONDS * 5,
```
Comment (vincentwschau) [inline indexer/services/roundtable/src/config.ts:142]: nit: As discussed offline, update to 10 seconds.

File: indexer/services/roundtable/src/tasks/cache-orderbook-mid-prices.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    });
+    await OrderbookMidPricesCache.fetchAndCacheOrderbookMidPrices(redisClient, marketTickers);
+  } catch (error) {
+    logger.error({
+      at: 'cache-orderbook-mid-prices#runTask',
+      message: error.message,
+      error,
+    });
+  }
+}
```
Comment (vincentwschau) [inline indexer/services/roundtable/src/tasks/cache-orderbook-mid-prices.ts:50]: nit: types.

File: indexer/packages/redis/__tests__/caches/orderbook-mid-prices-cache.test.ts [hunk 5]
```diff
@@ -28,109 +54,245 @@ describe('orderbook-mid-prices-cache', () => {
...
+
+      const result = await getMedianPrices(client, [defaultTicker]);
+      expect(result).toEqual({ 'BTC-USD': '50500' });
     });
 
-    it('returns the correct median price after 5 seconds', async () => {
+    it('returns the correct median price after 30 seconds', async () => {
       jest.useFakeTimers();
+      // Mock the getOrderBookMidPrice function for the ticker
+      const mockPrices = ['50000', '51000', '49000', '48000', '52000', '53000'];
```
Comment (vincentwschau) [inline indexer/packages/redis/__tests__/caches/orderbook-mid-prices-cache.test.ts:133]: nit: type

File: indexer/packages/redis/__tests__/caches/orderbook-mid-prices-cache.test.ts [hunk 6]
```diff
@@ -28,109 +54,245 @@ describe('orderbook-mid-prices-cache', () => {
...
+      client.zrange(`${ORDERBOOK_MID_PRICES_CACHE_KEY_PREFIX}${defaultTicker}`,
+        0,
+        -1,
+        (err: Error, res: string[]) => {
+          expect(res).toHaveLength(4);
+        });
+      expect(OrderbookLevelsCache.getOrderBookMidPrice).toHaveBeenCalledTimes(6);
+
+      // Check the median price
+      const result = await getMedianPrices(client, [defaultTicker]);
```
Comment (vincentwschau) [inline indexer/packages/redis/__tests__/caches/orderbook-mid-prices-cache.test.ts:163]: nit: type

**Review by vincentwschau — APPROVED** [review]
Logic looks good, nits around missing types for most variables.

**Review by teddyding — APPROVED** [review]

**Comment by adamfraser** [comment]
@mergify backport release/indexer/v7.x

**Pull request merged**